extras/kms/README.md

```diff
@@ -0,0 +1,88 @@
+  scope IDs and then associate a set of keys with each org and project within
+  your domain. 
+
+- `root key`:  the KEKs (keys to encrypt keys) of the system.  
+
+- `data key`:  the DEKs (keys to encrypt data) of the system and must have a
+  parent root key and a purpose.  
+
+- `purpose`:  Each data key (DEK) must have a one purpose.  For
+  example, you could define a purpose of `client-secrets` for a DEK that will be
+  used encrypt/decrypt wrapper operations on `client-secrets`
+  
+
+<hr>
+
+### Database Schema
+
+You'll find the database schema within the migrations directory.
+Currently postgres and sqlite are supported.  The implementation does make some
+use of triggers to ensure some of its data integrity. 
+
+The migrations are intended to be incorporated into your existing go-migrate
+migrations.  Feel free to change the migration file names, as long as they are
+applied in the same order as defined here.  FYI, the migrations include
+`kms_version` table which is used to ensure that the schema and module are
+compatible. 
+
+```
+High-level ERD                                          
+                                                          
+                                                          
+             ┌───────────────────────────────┐            
+             │                               ○            
+             ┼                               ┼            
+┌────────────────────────┐      ┌────────────────────────┐
+│      kms_root_key      │      │      kms_data_key      │
+├────────────────────────┤      ├────────────────────────┤
+│private_id              │      │private_id              │
+│scope_id                │      │root_key_id             │
+│                        │      │purpose                 │
+└────────────────────────┘      │                        │
+             ┼                  └────────────────────────┘
+             │                               ┼            
+             │                               │            
+             │                               │            
+             │                               │            
+             ┼                               ┼            
+            ╱│╲                             ╱│╲           
+┌────────────────────────┐      ┌────────────────────────┐
+│  kms_root_key_version  │      │  kms_data_key_version  │
+├────────────────────────┤      ├────────────────────────┤
+│private_id              │      │private_id              │
+│root_key_id             │      │data_key_id             │
+│key                     │      │root_key_id             │
+│version                 │      │key                     │
+│                        │      │version                 │
+└────────────────────────┘      └────────────────────────┘
+             ┼                               ┼            
+             │                               ○            
+             └───────────────────────────────┘               
+          
+```
```

Comment:
FYI github has support for mermaid diagrams if we want to prettify this: https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/

Reply:
ty but personally, I like the txt version